ZENKO-3106 add release stage (#3318)
scality/cloudserver: docs/RELEASE.md

# Cloudserver Release Plan

> ## Docker Image generation

Reviewer: ```suggestion
## Docker Image Generation
```

Docker images are hosted on [registry.scality.com](registry.scality.com).

> It has two names spaces for cloudserver:

Reviewer: ```suggestion
CloudServer has two namespaces there:
```

* Production Namespace: registry.scality.com/cloudserver
* Dev Namespace: registry.scality.com/cloudserver-dev

> The CI will push images with every CI build tagging the

Reviewer: ```suggestion
With every CI build, the CI will push images, tagging the
```

> content with the developer’s branch short SHA-1 commit hash.

Reviewer: ```suggestion
content with the developer branch's short SHA-1 commit hash.
```

This allows those images to be used by developers, CI builds,
build chain and so on.

Tagged versions of cloudserver will be stored in the production namespace.

> ## How to pull docker images

Reviewer: ```suggestion
How to Pull Docker Images
```

Author: Any reason to remove the `##` ?

Author: ```suggestion
## How to Pull Docker Images
```

```sh
docker pull registry.scality.com/cloudserver-dev/cloudserver:<commit hash>
docker pull registry.scality.com/cloudserver/cloudserver:<tag>
```

## Release Process

To release a production image:

> * Chose the name of the tag for the repository and the docker image.

Reviewer: ```suggestion
* Name the tag for the repository and Docker image.
```

> * Update the `package.json` using the command `yarn version` with the same tag.

Reviewer: ```suggestion
* Use the `yarn version` command with the same tag to update `package.json`.
```

* Create a PR and merge the `package.json` change.

* Tag the repository using the same tag.

> * With the below parameters, [force a build](https://eve.devsca.com/github/scality/cloudserver/#/builders/bootstrap/force/force):

Reviewer: ```suggestion
* [Force a build](https://eve.devsca.com/github/scality/cloudserver/#/builders/bootstrap/force/force) using: 
```

>   * A given branch that ideally match the tag.

Reviewer: ```suggestion
  * A given branch that ideally matches the tag.
```

>   * Use the `release` stage.

Reviewer: ```suggestion
  * The `release` stage.
```

>   * Extra property with name as `tag` and the value as the actual tag.

Reviewer: ```suggestion
  * An extra property with the name `tag` and its value being the actual tag.
```
